lsql/judge/templates/problem_trigger.html

```diff
@@ -4,7 +4,22 @@ <h2>Sentencias ejecutadas</h2>
 <pre><code>{{ problem.tests }}</code></pre>
 
 <h2>Resultado esperado</h2>
-{% for name, table in problem.expected_result.items %}
+{% if problem.show_added %}
```

Comment:
Mejor un único bucle que recorra todas y muestre únicamente aquellas que deba, siguiendo la visión de no agrupar por tipos sino de añadir un pequeño mensaje al lado del nombre.

Este código es el mismo en las 3 plantillas y tiene varias líneas, me imagino que es hijo de Ctrl-C + Ctrl-V. **Crea una macro de plantilla e invócala en las 3 plantillas para tener una única copia del código y estar seguros de que es la misma en los 3 usos**

Reply:
¿Te refieres a que se muestre así y no con un encabezado?

![TablaAñadida](https://user-images.githubusercontent.com/48255774/110337897-b17ffe00-8026-11eb-98f3-bb4dfcfaabe7.png)

Reply:
Las tablas eliminadas que no están en problem.expected_results sí que tengo que hacer otro for para ellas ¿no?

Reply:
Además no entiendo bien creo, ¿quieres que haga un bucle que recorra todas las tablas, tenga o no que mostrarlas, y ya dentro del bucle decido si las muestro o no? Eso sería iterar más veces de las necesarias creo yo.

Reply:
Perdón por la cantidad de mensajes, pero según voy avanzando voy teniendo más dudas. ¿Con lo de crear una macro de plantilla te refieres a un HTML nuevo que extienda de Problem.html y despues Problem_proc.html extienda de esta no?

Reply:
> Además no entiendo bien creo, ¿quieres que haga un bucle que recorra todas las tablas, tenga o no que mostrarlas, y ya dentro del bucle decido si las muestro o no? Eso sería iterar más veces de las necesarias creo yo.

Como prefieras, si lo ves más complejo elige la opción que más te guste.

Reply:
> Perdón por la cantidad de mensajes, pero según voy avanzando voy teniendo más dudas. ¿Con lo de crear una macro de plantilla te refieres a un HTML nuevo que extienda de Problem.html y despues Problem_proc.html extienda de esta no?

No, me refiero a una macro como en https://github.com/emartinm/lsql/blob/master/lsql/judge/templates/show_table.html, para luego usarla dentro de las plantillas para Proc, Trigger y DML como en https://github.com/emartinm/lsql/blob/5ae4aa21ac67ed040713a9a30d74c867d9bf6cf3/lsql/judge/templates/problem_proc.html#L8. Mejor no modificar la jerarquía de plantillas, que yo creo que ya tiene bastantes niveles.

Reply:
> ¿Te refieres a que se muestre así y no con un encabezado?
> 
> ![TablaAñadida](https://user-images.githubusercontent.com/48255774/110337897-b17ffe00-8026-11eb-98f3-bb4dfcfaabe7.png)

Sí, mi idea era algo así, aunque pondría el texto un poco más pequeño que el título de la tabla, entre paréntesis y quizá con un estilo diferente para hacer una diferencia visual entre el nombre y el mensaje informativo.


> Las tablas eliminadas que no están en problem.expected_results sí que tengo que hacer otro for para ellas ¿no?

Sí, o recorrer la unión de `initial U expected`

Reply:
Aquí lo único que no he sabido como hacer es la letra más pequeña.

Reply:
¿Pero si has sido capaz de hacer que tenga un estilo diferente (itálica, negrita, etc.)? ¿Puedes incluir una nueva captura para verlo?

Reply:
No, solo he podido ponerle paréntesis.

![image](https://user-images.githubusercontent.com/48255774/110480594-f23f4c00-80e6-11eb-906b-d319a5e2ed8a.png)

No sé enlazar partes del código, pero esto lo hago en Django al llamar a show_table con 
`name = name|add:" (Tabla modificada)"`